### PR TITLE
Multi-speaker VITS & Hi-Fi TTS dataset structure

### PR DESCRIPTION
**Pull request by zyingt**
## ✨ Description

This PR introduces multi-speaker support for the current VITS model. It allows for the synthesis of speech in multiple voices and enables users to choose the specific speaker's voice that suits their preferences. To test this PR, you may follow the guidelines specified in the latest **egs/tts/VITS/README.md**.

## 🚧 Related Issues

None

## 👨‍💻 Changes Proposed

**[1] Enabling multi-speaker VITS support:**
- Updated **egs/tts/VITS/run.sh**, **exp_config.json** and **README.md** to include necessary arguments and instructions for enabling multi-speaker training and inferencing in VITS
- Included `intersperse` function in **utils/data_utils.py**, allowing the insertion of blanks (0) within consecutive phone IDs to regulate speaking speed. 

**[2] Streamlined Hi-Fi TTS dataset preprocessing:**
- Introduced the Hi-Fi TTS dataset structure in **egs/datasets/README.md**
- Updated **preprocessors/processor.py** to accommodate the Hi-Fi TTS preprocessor 

**[3] Changes on VITS dataset loader:**
- Included metadata filter in **models/tts/vits/vits_dataset.py** to exclude very short segments such that `frame_len < self.cfg.preprocess.segment_size // self.cfg.preprocess.hop_size`
- Shifted variable declaration of `processed_data_dir` from class `VITSTestDataset(TTSTestDataset)` in **models/tts/vits/vits_dataset.py** out of ifs condition as it has been referenced within `elif cfg.preprocess.use_phone: ` (line 88, latest) without prior declaration.

**[4] Enhance model compatibility for different accelerate versions**
- The Hi-fi TTS VITS checkpoint was trained on accelerate v0.25, the resulting model file is `model.safetensors` instead of `pytorch_model.bin`. To enable users to use the checkpoint successfully, **models/tts/base/tts_inferece.py** is modified to add another way of loading model when users' accelerate version is <0.25.

**[5] Black formatting**

## 🧑‍🤝‍🧑 Who Can Review?

@lmxue @RMSnow 

## 🛠 TODO

- Test multi-speaker VITS pipeline (preprocessing->feature extraction->training->resume training->inference for single and batch) on Hi-Fi TTS (Done)
- Test single-speaker VITS pipeline (preprocessing->feature extraction->training->resume training->inference for single and batch) on LJSpeech (Done)

## ✅ Checklist

- [ ]  Code has been reviewed
- [ ]  Code complies with the project's code standards and best practices
- [ ]  Code has passed all tests
- [ ]  Code does not affect the normal use of existing features
- [ ]  Code has been commented properly
- [ ]  Documentation has been updated (if applicable)
- [ ]  Demo/checkpoint has been attached (if applicable)

### CODE REVIEW

File: egs/tts/VITS/README.md
Comment (RMSnow) [inline egs/tts/VITS/README.md:1]: Can Multi-speaker VITS not be applied to the Single-speaker version? In other words, is it necessary to write both "single speaker VITS" section and "multi-speaker VITS" section? Can they be merged?

File: utils/data_utils.py [hunk 1]
```diff
@@ -12,6 +12,12 @@
 from sklearn.preprocessing import StandardScaler
 
 
+def intersperse(lst, item):
```
Comment (RMSnow) [inline utils/data_utils.py:15]: Add comments for this function. For example, what is it for?

**Review by RMSnow — CHANGES_REQUESTED** [review]

File: egs/tts/VITS/README.md
Comment (lmxue) [inline egs/tts/VITS/README.md:1]: The current recipe includes instructions for both single-speaker and multi-speaker setups, with many duplications. We can write this recipe as a multi-speaker VITS. It also works for single-speaker VITS  when turning off the multi-speaker-related switches set in the `config.json`.

File: models/tts/base/tts_dataset.py [hunk 2]
```diff
@@ -209,6 +209,9 @@ def __init__(self, cfg, dataset, is_valid=False):
                     phon_id_collator = phoneIDCollation(cfg, dataset=dataset)
                     sequence = phon_id_collator.get_phone_id_sequence(cfg, phones_seq)
 
+                    if cfg.preprocess.add_blank and cfg.train.multi_speaker_training:
```
Comment (lmxue) [inline models/tts/base/tts_dataset.py:212]: The condition `cfg.train.multi_speaker_training` can be removed, allowing the `intersperse` function to also be applicable to single-speaker training.

File: models/tts/vits/vits_inference.py [hunk 3]
```diff
@@ -120,6 +121,9 @@ def inference_for_single_utterance(
         )
         phone_id_seq = phon_id_collator.get_phone_id_sequence(self.cfg, phone_seq)
 
+        if self.cfg.preprocess.add_blank and self.cfg.train.multi_speaker_training:
```
Comment (lmxue) [inline models/tts/vits/vits_inference.py:124]: The condition `cfg.train.multi_speaker_training` can be removed, allowing the `intersperse` function to also be applicable to single-speaker training.

**Review by lmxue — CHANGES_REQUESTED** [review]

File: egs/tts/VITS/README.md
Comment (zyingt) [reply egs/tts/VITS/README.md:1]: Yes, they have been merged. Here's the latest version: https://github.com/open-mmlab/Amphion/pull/131/files#diff-83c5cc63dddb1e8db2ded5bf8c8bab9846ffaa59b1b475f2fb94ff99a28f1b73. Currently, it delivers instructions for single-speaker version with additional instructions for multi-speaker purposes, those who want to train multi-speaker VITS should follow the additional instructions given in each section to turn on the multi-speaker related args. While for the inference part, it is split into single and multi-speaker section, for the ease of users in copying the entire code blocks with minimal editing.

File: utils/data_utils.py (hunk 1 above)
Comment (zyingt) [reply utils/data_utils.py:15]: Added, please check the latest version and let me know if there's further revision required: https://github.com/open-mmlab/Amphion/pull/131/files#diff-d8373728b78ea7f07bc4558415949336dd3efbe3ca55c04371357a5ea37895bd

File: egs/tts/VITS/README.md
Comment (zyingt) [reply egs/tts/VITS/README.md:1]: Here's the latest version: https://github.com/open-mmlab/Amphion/pull/131/files#diff-83c5cc63dddb1e8db2ded5bf8c8bab9846ffaa59b1b475f2fb94ff99a28f1b73. Currently, it serves as single-speaker recipe that also works for multi-speaker VITS by turning on switches related to multi-speaker setting in `exp_config.json`.

File: models/tts/vits/vits_inference.py (hunk 3 above)
Comment (zyingt) [reply models/tts/vits/vits_inference.py]: Got it, removed.

**Review by RMSnow — CHANGES_REQUESTED** [review]
Use black to format the code.

File: egs/tts/VITS/README.md [hunk 4]
```diff
@@ -67,11 +76,16 @@ sh egs/tts/VITS/run.sh --stage 1
...
 
 We provide the default hyparameters in the `exp_config.json`. They can work on single NVIDIA-24g GPU. You can adjust them based on your GPU machines.
+For training multi-speaker TTS model, specify the `n_speakers` according to the number of speakers in your dataset(s) and set `multi_speaker_training` to `true`.
 
-```
-"train": {
-        "batch_size": 16,
-    }
+```json
+  "model": {
```
Comment (RMSnow) [inline egs/tts/VITS/README.md:82]: I think 

```json
"model": {
    "n_speakers": 1 //Number of speakers in the dataset(s) used. The default value is 1 if not specified.
  },
```
may be better. `0` is strange.

Comment (RMSnow) [inline egs/tts/VITS/README.md:82]: I think 

```json
"model": {
    "n_speakers": 1 //Number of speakers in the dataset(s) used. The default value is 1 if not specified.
  },
```
may be better. Using `0` as default value is strange.

**Review by RMSnow — CHANGES_REQUESTED** [review]

File: egs/tts/VITS/README.md (hunk 4 above)
Comment (lmxue) [reply egs/tts/VITS/README.md:82]: For the current model structure, setting `n_speaker=0` as the default value allows for more flexible control over whether to use speaker embedding in a single speaker scenario, as well as the size of the speaker embedding. The specific code can be found [here](https://github.com/open-mmlab/Amphion/blob/1c4a9afd0465f0d1318b65c3268bca0230005206/models/tts/vits/vits_trainer.py#L26) and [here](https://github.com/open-mmlab/Amphion/blob/9a743bed2d50412c6715197abec05ae12d3ed2b5/models/tts/vits/vits.py#L247C12-L247C22).

Of course, setting `n_speaker=1` as the default value also works. But it requires modifications in multiple sections of the code, which comes at a high cost.

**Review by lmxue — APPROVED** [review]
It looks good now.

**Review by RMSnow — APPROVED** [review]

**Pull request merged**